src/blocks/weather/nws.rs

```diff
@@ -0,0 +1,380 @@
+#[serde(rename_all = "camelCase")]
+struct ApiPointsProperties {
+    forecast_hourly: String,
+    relative_location: ApiRelativeLocation,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct ApiRelativeLocation {
+    properties: ApiRelativeLocationProperties,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct ApiRelativeLocationProperties {
+    city: String,
+    state: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct ApiForecastResponse {
+    properties: ApiForecastProperties,
+}
+
+#[derive(Deserialize, Debug)]
+struct ApiForecastProperties {
+    periods: Vec<ApiForecast>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct ApiValue {
+    value: f64,
+    unit_code: String,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct ApiForecast {
+    is_daytime: bool,
+    temperature: ApiValue,
+    relative_humidity: ApiValue,
+    wind_speed: ApiValue,
+    wind_direction: String,
+    short_forecast: String,
+}
+
+impl ApiForecast {
+    fn wind_direction(&self) -> f64 {
+        let dir = match self.wind_direction.as_str() {
+            "NNE" => 1,
+            "NE" => 2,
+            "ENE" => 3,
+            "E" => 4,
+            "ESE" => 5,
+            "SE" => 6,
+            "SSE" => 7,
+            "S" => 8,
+            "SSW" => 9,
+            "SW" => 10,
+            "WSW" => 11,
+            "W" => 12,
+            "WNW" => 13,
+            "NW" => 14,
+            "NNW" => 15,
+            _ => 0,
+        };
+        (dir as f64) * (360.0 / 16.0)
+    }
+
+    fn icon_to_word(icon: WeatherIcon) -> String {
+        match icon {
+            WeatherIcon::Clear { .. } => "Clear",
+            WeatherIcon::Clouds { .. } => "Clouds",
+            WeatherIcon::Fog { .. } => "Fog",
+            WeatherIcon::Thunder { .. } => "Thunder",
+            WeatherIcon::Rain { .. } => "Rain",
+            WeatherIcon::Snow => "Snow",
+            WeatherIcon::Default => "Unknown",
+        }
+        .to_string()
+    }
+
+    fn wind_kmh(&self) -> f64 {
+        if self.wind_speed.unit_code.ends_with("km_h-1") {
+            self.wind_speed.value
+        } else {
+            self.wind_speed.value * MPH_TO_KPH
+        }
+    }
+
+    fn apparent_temp(&self) -> f64 {
+        let temp = if self.temperature.unit_code.ends_with("degC") {
+            self.temperature.value
+        } else {
+            (self.temperature.value - 32.0) * 5.0 / 9.0
+        };
+        let humidity = self.relative_humidity.value;
+        let wind_speed = self.wind_kmh();
+        australian_apparent_temp(temp, humidity, wind_speed)
+    }
+
+    fn to_moment(&self) -> WeatherMoment {
+        let icon = short_forecast_to_icon(&self.short_forecast, !self.is_daytime);
+        let weather = Self::icon_to_word(icon);
+        WeatherMoment {
+            icon: short_forecast_to_icon(&self.short_forecast, !self.is_daytime),
```

Comment:
Good point! And I don't even need to switch to a reference since `WeatherIcon` is `Copy`. Extra call has been dropped.

Reply:
It's better to not copy if it's not necessary

Reply:
I'm not sure I understand why. `WeatherIcon` is 2 bytes in size and can get passed in directly as an argument in registers, versus following a reference. I certainly get wanting to avoid `clone`, but `copy`?

Reply:
Sorry, I think you're right. 😅 I hadn't looked again at what a `WeatherIcon` and for some reason in my head though it was larger.

Reply:
Totally fair, I certainly had to double-check it myself too.